build.sbt

```diff
@@ -71,13 +71,17 @@ lazy val rules = project
   .in(file("scalafix-rules"))
   .settings(
     moduleName := "scalafix-rules",
+    buildInfoKeys ++= Seq[BuildInfoKey](
+      "supportedScalaVersionforRules" -> supportedScalaVersionforRules
+    ),
     description := "Built-in Scalafix rules",
     libraryDependencies ++= List(
       "org.scalameta" % "semanticdb-scalac-core" % scalametaV cross CrossVersion.full,
       collectionCompat
     )
   )
   .dependsOn(core)
+  .enablePlugins(BuildInfoPlugin)
```

Comment:
did you confirm the file was generated in a package that wouldn't potentially conflict with other build infos depending on the classpath loading?

Reply:
No, and actually I dont know how to verify that. 
In fact reflect dependsOn core and rules, which both contains BuildInfo. I will change buildInfo object name .